src/biomappings/wsgi.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+
+"""Web curation interface for :mod:`biomappings`."""
+
+import getpass
+from typing import Optional
+
+import flask
+import flask_bootstrap
+
+from biomappings.resources import append_false_mappings, append_true_mappings, load_predictions, write_predictions
+from biomappings.utils import commit
+
+app = flask.Flask(__name__)
+flask_bootstrap.Bootstrap(app)
+
+# A mapping from your computer's user, returned by getuser.getpass()
+KNOWN_USERS = {
+    'cthoyt': '0000-0003-4423-4370',
+}
+
+
+class Controller:
+    """A module for interacting with the predictions and mappings."""
+
+    def __init__(self):  # noqa: D107
+        self._predictions = load_predictions()
+        self._marked = {}
+        self.total = 0
+
+    def predictions(self, top: Optional[int] = None):
+        """Iterate over predictions.
+
+        :param top: If given, only iterate this number of predictions.
+        """
+        if top is None:
+            for i, p in enumerate(self._predictions):
+                if i not in self._marked:
+                    yield i, p
+        else:
+            counter = 0
+            for i, p in enumerate(self._predictions):
+                if i not in self._marked:
+                    counter += 1
+                    yield i, p
+                if counter > top:
+                    break
+
+    def mark(self, i: int, correct: bool):
+        """Mark the given equivalency as correct."""
+        if i not in self._marked:
+            self.total += 1
+        self._marked[i] = correct
+
+    def persist(self):
+        """Save the current markings to the source files."""
+        curated_true_entries = []
+        curated_false_entries = []
+
+        for i, correct in self._marked.items():
```

Comment:
possible bug - since below there will be popping, this either should be iterated from big to small, or the prediction list should be rebuild in a different place